### ISSUE TITLE
query.ImOnline.authoredBlocks always returns 0 for old session indexes. 

### ISSUE THREAD

**Issue by dillu24** [polkadot-js/api#1661]
I am running an application where I need to check whether a validator has authored blocks in previous sessions. I am doing this using the `query.ImOnline.authoredBlocks` function, however, I have noticed that when calling this function on previous sessions it always returns 0 even for validators that I am sure that they authored blocks in that session. 

Is this by design or just a bug?

**Comment by jacogr** [comment]
This operates exactly as intended. The state is not a database, it keep only the last information required as for operation. So when you query older sessions, it is removed.

On archive nodes you can return the information with at queries, at in-session blocks.

TL;DR The state gets cleared as sessions tick over. As designed to not make the state bloat.

**Comment by jacogr** [comment]
More relevant info with background - https://github.com/polkadot-js/apps/issues/1954
